src/components/com_tjcertificate/administrator/controllers/trainingrecord.php

```diff
@@ -0,0 +1,229 @@
+		// Validate the posted data.
+		$form = $model->getForm($data, false);
+
+		if (!$form)
+		{
+			throw new \Exception($model->getError(), 500);
+		}
+
+		$validData = $model->validate($form, $data);
+
+		// Check for validation errors.
+		if ($validData === false)
+		{
+			// Get the validation messages.
+			$errors = $model->getErrors();
+
+			if (!empty($errors))
+			{
+				// Push up to three validation messages out to the user.
+				for ($i = 0, $n = count($errors); $i < $n && $i < 3; $i++)
+				{
+					if ($errors[$i] instanceof Exception)
+					{
+						$app->enqueueMessage($errors[$i]->getMessage(), 'warning');
+					}
+					else
+					{
+						$app->enqueueMessage($errors[$i], 'warning');
+					}
+				}
+			}
+
+			// Save the data in the session.
+			$app->setUserState('com_tjcertificate.edit.trainingrecord.data', $data);
+
+			// Redirect back to the edit screen.
+			$this->setRedirect(Route::_('index.php?option=com_tjcertificate&view=trainingrecord&layout=edit&id=' . $recordId, false));
+
+			return false;
+		}
+
+		if ($validData['assigned_user_id'])
+		{
+			$validData['user_id'] = $validData['assigned_user_id'];
+		}
+		else
+		{
+			$validData['user_id'] = $user->id;
+		}
+
+		$validData['client'] = "external";
+		$validData['state'] = $validData['state'] ? $validData['state'] : "-1";
+		$validData['is_external'] = 1;
+
+		$file = $app->input->files->get('jform', array(), 'array');
+
+		if (!empty($file['cert_file']))
+		{
+			$validData['old_media_ids'] = $app->input->get('oldFiles', 0, 'INT');
+			$uploadData = $model->uploadMedia($file, $validData);
+			$validData['cert_file'] = $uploadData['source'];
+		}
+
+		$certificateModel = TJCERT::model('Certificate', array('ignore_request' => true));
+
+		$certificateModel->save($validData);
+
+		$modelMediaXref = TJMediaXref::getInstance();
+
+		if ($uploadData['id'])
+		{
+			$mediaData['id'] = '';
+			$mediaData['client_id'] = $certificateModel->getState('certificate.id');
+			$mediaData['media_id'] = $uploadData['id'];
+			$mediaData['client'] = 'com_tjcertificate';
+			$modelMediaXref->bind($mediaData);
+			$modelMediaXref->save();
+		}
+
+		$this->setMessage(Text::_('COM_TJCERTIFICATE_TRAINING_RECORD_SAVE_SUCCESSFULLY'));
+
+		// Redirect back to the edit screen.
+		$this->setRedirect(
+			Route::_('index.php?option=com_tjcertificate&view=trainingrecord&layout=edit&id=' . $certificateModel->getState('certificate.id'), false)
+			);
+
+		// Flush the data from the session.
+		$app->setUserState('com_tjcertificate.edit.trainingrecord.data', null);
+	}
+
+	/**
+	 * Cancel operation
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function cancel()
+	{
+		// Check for request forgeries.
+		$this->checkToken('request');
+
+		// Clear data from session.
+		\JFactory::getApplication()->setUserState('com_tjcertificate.edit.trainingrecord.data', null);
+
+		$this->setRedirect(Route::_('index.php?option=com_tjcertificate&view=certificates&layout=my', false));
```

Comment:
This is the admin's controller, there is no view like `layout=my` @tushar33

Reply:
cancel is using only in the frontend